src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,297 @@
+        join(pluginTranslationPath, 'de.json'),
+        join(pluginTranslationPath, 'en.json')
+      ];
+      translationFiles.forEach(i18n.registerTranslations);
+    });
+
+    afterEach('unregisterTranslations - one plugin', function () {
+      i18n.unregisterTranslations();
+    });
+
+    describe('getTranslationsForLocale', function () {
+
+      it('should return the translations for en locale as registered' , function () {
+        const locales = [
+          { code: 'en', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!'
+        };
+        return checkTranslations(locales, expectedTranslationJson);
+      });
+
+      it('should return the translations for de locale as registered' , function () {
+        const locales = [
+          { code: 'de', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        return checkTranslations(locales, expectedTranslationJson);
+      });
+
+      it('should return an empty object for locale with no translations' , function () {
+        const locales = [
+          { code: 'es', region: undefined, quality: 0.8 }
+        ];
+        return checkTranslations(locales, {});
+      });
+
+    });
+
+    describe('getRegisteredTranslationLocales', function () {
+
+      it('should return all locales that are registered' , function () {
+        const expectedLocales = ['en', 'de'];
+        return checkRegisteredLocales(expectedLocales);
+      });
+
+    });
+
+    describe('getTranslationsForPriorityLocaleFromLocaleList', function () {
+
+      it('should first return empty object for no locale registered and ' +
+         'then return translations for default locale as requested' , function () {
+        const locales = [
+          { code: 'es', region: 'ES', quality: 1.0 },
+          { code: 'fr', region: undefined, quality: 0.8 }
+        ];
+        const expectedTranslations = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!'
+        };
+        const server = {
+          config: {
+            get: function () {
+              return 'en';
+            }
+          }
+        };
+        i18n.setI18nConfig(server.config);
+        checkTranslations(locales, {});
+        return checkTranslationsForDefaultLocale(expectedTranslations);
```

Comment:
`getTranslations` isn't using the default locale under the hood. When the requested languages aren't found, it returns an empty object. Which I think is the correct behavior.